File: java/05-04-2024/morning/ass3/Pati/InHousePatient.java
import Pati.Patient;

public class InHousePatient extends Patient{
    
    private int discount;

    public int getDiscount(){
        return discount;
    }

    public void setDiscount(int dis){
        discount = dis;
    }

    public double getBillAmount(){
        if(noOfDays * getPricePerDay() > 5000){
            return noOfDays * getPricePerDay() * discount;
        }
        return noOfDays * getPricePerDay() * 0.05;
     }
}

File: dotnet/assignments/assignment2/InHousePatient.cs
class InHousePatient : Patient
{
    public double discount { get; set; }

    // discount = getBillAmount() > 5000 ? discount : 0.5;

    public override double getBillAmount(){
        double amount = base.getBillAmount();
        if(base.getBillAmount() < 5000){
            discount = 0.05;
        }
        amount *= discount;
        return base.getPricePerDay() * noOfDays - amount;
    }
}
